ci/benchmark_bencher.yml

```
@@ -0,0 +1,55 @@
+include:
+  - local: 'ci/base.yml'
+
+.benchmark_model_stencils:
+  stage: benchmark
+  script:
+    - curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+    - source "$HOME/.cargo/env"
+    - cargo install --git https://github.com/finkandreas/bencher --branch main --locked --force bencher_cli # Install bencher CLI
+    - uv sync --no-dev --extra=dace --extra=io --extra=testing --extra=$ICON4PY_NOX_UV_CUSTOM_SESSION_EXTRAS --group=test
+    - source .venv/bin/activate
+    - PR_ID=$(echo "${CI_COMMIT_BRANCH}" | grep -o 'pr[0-9]*' | grep -o '[0-9]*')
+    - FEATURE_BRANCH=$(curl -s https://api.github.com/repos/C2SM/icon4py/pulls/$PR_ID | jq -r '.head.ref')
+    - export GITHUB_EVENT_NAME=pull_request # needed for GitHub PR comments - along with --github-actions -
+    - export GITHUB_STEP_SUMMARY=$CI_PROJECT_DIR/step_summary.log # needed for GitHub PR comments - along with --github-actions -
+    - export GITHUB_SHA=$CI_COMMIT_SHA # needed for GitHub PR comments - along with --github-actions -
+    - export GITHUB_ACTIONS=true # needed for GitHub PR comments - along with --github-actions -
+    - export GITHUB_EVENT_PATH=$CI_PROJECT_DIR/event.json # needed for GitHub PR comments - along with --github-actions -
+    - | # needed for GitHub PR comments - along with --github-actions -
+      echo "{\"pull_request\": {\"head\": {\"repo\": {\"full_name\": \"C2SM/icon4py\"}}}, \"repository\": {\"full_name\": \"C2SM/icon4py\"}, \"number\": $PR_ID}" > $CI_PROJECT_DIR/event.json
+    - | # Run the benchmarks
+      bencher run \
+      --project "$BENCHER_PROJECT" \
+      --token "$BENCHER_API_TOKEN" \
+      --github-actions $GD_COMMENT_TOKEN \
+      --branch "$FEATURE_BRANCH" \
+      --start-point main \
```

Here instead you could try:
```suggestion
      --start-point $CI_MERGE_REQUEST_TARGET_BRANCH_NAME \
```